analyze-trace.ts

```diff
@@ -162,18 +193,41 @@ async function main(): Promise<void> {
         }
     }
 
-    await printHotStacks(root);
+    await makeHotStacks(root);
+}
+
+type TreeNode = {
```

Comment:
`interface`?

Reply:
Oh, maybe that doesn't work with recursive types?

Reply:
interface will be fine, they can recurse too